### PR TITLE
feat(pkg): AUR package is changed to reflect packaging guidelines

### LINKED ISSUE #243: Correct PKGBUILD for AUR release
The PKGBUILD for the AUR release should not use `sudo` commands. 

Currently the PKGBUILD used for AUR works but uses sudo commands. This is "not allowed" and subpar behaviour.

OUTCOME:
The PKGBUILD for the AUR has no sudo and works as expected. 

[For reference see eg this PKGBUILD](https://github.com/archlinux/svntogit-community/blob/packages/jless/trunk/PKGBUILD)

### PR DESCRIPTION
**Pull request by iamwacko**
This will fix #243 . The PKGBUILD now builds siera-cli and doesn't do any of the weird sudo curl stuff. The runner is also now an archlinux container, so it can have checksums and such. 

This means the siera AUR package is no longer a binary package, but I have made and uploaded a siera-bin (made properly according to package guidelines) basically does what the old siera package used to do. I could put that in the build process and hand over the package, if you want me to.

### CODE REVIEW

**Comment by TimoGlastra** [comment]
Hey @iamwacko, thanks for looking into this and opening a PR. @berendsliedrecht is away for the week, he'll take a look at this when he's back next week

**Comment by TimoGlastra** [comment]
@berendsliedrecht bump on this one

File: pkg/PKGBUILD [hunk 1]
```diff
@@ -1,17 +1,37 @@
...
+source=("https://github.com/animo/siera-cli/archive/refs/tags/v$pkgver.tar.gz")
+depends=('cargo')
 provides=('siera')
-md5sums=()
+sha256sums=()
+
+prepare() {
+	cd "siera-cli-$pkgver"
+	case $CARCH in
+		"pentium4")
```
Comment (berendsliedrecht) [inline pkg/PKGBUILD:17]: Is this the proper check for this? Shouldn`t we use `arch`?

**Review by berendsliedrecht — APPROVED** [review]
Just one comment regarding the `$CARCH` check.

Thanks for the contribution!

File: pkg/PKGBUILD (hunk 1 above)
Comment (iamwacko) [reply pkg/PKGBUILD:17]: technically yeah, but with [archlinux32](https://archlinux32.org/) the $CARCH isn't usually i686, so it wouldn't properly compile. I have a few computers running archlinux32, so it's a habit I've picked up.

**Comment by berendsliedrecht** [comment]
@iamwacko could you rebase with main? `audit` is failing but that is fixed now in `main. Thanks!

**Comment by iamwacko** [comment]
> @iamwacko could you rebase with main? `audit` is failing but that is fixed now in `main. Thanks!

Ok, done

**Pull request merged**